pilot/cmd/pilot-agent/main.go

```diff
@@ -450,6 +473,25 @@ func getDNSDomain(domain string) string {
 	return domain
 }
 
+// check if SDS UDS path and token path exist, if both exist, requests key/cert
+// using SDS instead of secret mount.
+func getSDSData() (bool, string) {
+	if !sdsEnabledVar.Get() {
+		return false, ""
+	}
+
+	// If sdsenabled env var is set but uds doesn't exist, treat sds as disabled.
+	if !waitForFile(sdsUDSPath, sdsUdsWaitTimeout) {
```

Comment:
Again - no wait needed (or wanted). Either node agent is running as daemon and file will be there and mounted - or it doesn't, in which case it should use the certs (or fail - and let k8s deal with the restart).

Reply:
I mean - look for certs, and if certs are not found fail ( or if user chooses go on without certs - in a future PR if needed)